tests/Mock/RepositoryMock.php

```diff
@@ -0,0 +1,53 @@
+<?php
+/**
+ * This file is part of the prooph/event-sourcing.
+ * (c) 2014-2016 prooph software GmbH <contact@prooph.de>
+ * (c) 2015-2016 Sascha-Oliver Prolic <saschaprolic@googlemail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ProophTest\EventSourcing\Mock;
+
+use Prooph\EventSourcing\Aggregate\AggregateRepository;
+use Prooph\EventSourcing\Aggregate\AggregateTranslator;
+use Prooph\EventSourcing\Aggregate\AggregateType;
+use Prooph\EventStore\EventStore;
+use Prooph\EventStore\Snapshot\SnapshotStore;
+use Prooph\EventStore\Stream\StreamName;
+
+final class RepositoryMock extends AggregateRepository
+{
+    public function accessEventStore(): EventStore
+    {
+        return $this->eventStore;
+    }
+
+    public function accessAggregateType(): AggregateType
+    {
+        return $this->aggregateType;
+    }
+
+    public function accessAggregateTranslator(): AggregateTranslator
+    {
+        return $this->aggregateTranslator;
+    }
+
+    public function accessDeterminedStreamName(?string $aggregateId = null): StreamName
```

Comment:
@oqq yes, the `?` is optional here but it's perfectly valid and it's is more explicit this way. https://wiki.php.net/rfc/nullable_types